src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,671 @@
+        />
+      </this.Wrapper>
+    )
+  }
+
+  private renderItems(
+    variables: ComponentVariablesInput,
+    getItemProps: (options: GetItemPropsOptions<DropdownListItem>) => any,
+    highlightedIndex: number,
+  ) {
+    const { items, noResultsMessage } = this.props
+    if (items.length > 0) {
+      return items.map((item, index) => {
+        // to avoid passing undefined for these props, will spread.
+        const optionalItemProps = {
+          media: item.image && <Image src={item.image} avatar />,
+          content: item.content,
+        }
+        return (
+          <ListItem
+            key={item.key || item.header}
+            header={item.header}
+            {...optionalItemProps}
+            variables={{
+              ...(highlightedIndex === index && {
+                headerColor: variables.listItemTextColor,
+                contentColor: variables.listItemTextColor,
+              }),
+            }}
+            styles={{
+              backgroundColor:
+                highlightedIndex === index
+                  ? variables.listItemHighlightedBackgroundColor
+                  : variables.listItemBackgroundColor,
+            }}
+            {...getItemProps({ index, item })}
+          />
+        )
+      })
+    }
+    // render no match message.
+    return [
+      {
+        key: 'people-picker-no-results-item',
+        content: (
+          <Text weight="bold" content={noResultsMessage || `We couldn't find any matches.`} />
+        ),
+        styles: {
+          backgroundColor: variables.listItemBackgroundColor,
+        },
+      },
+    ]
+  }
+
+  private renderActiveValues(styles: ComponentSlotStylesInput) {
+    const value = this.state.value as DropdownListItem[]
+    const { getA11yRemoveItemMessage } = this.props
+
+    return value.length === 0
+      ? null
+      : value.map((item, index) => {
+          const optionalImage = {
+            ...(item.image && {
+              image: { src: item.image, avatar: true, onClick: this.handleLabelClick },
+            }),
+          }
+          return (
+            <Label
+              role="presentation"
+              styles={styles.activeListLabel}
+              circular
+              key={`active-item-${item.key || item.header}`}
+              content={<Text content={item.header} onClick={this.handleLabelClick} />}
+              {...optionalImage}
+              icon={{
+                name: 'close',
+                onClick: this.handleCloseIconClick.bind(this, item),
+                onKeyDown: this.handleCloseIconKeyDown.bind(this, item),
+                'aria-label': getA11yRemoveItemMessage
+                  ? getA11yRemoveItemMessage(item)
+                  : `Remove ${item.header} from selection.`,
+                'aria-hidden': false,
+                role: 'button',
+              }}
+            />
+          )
+        })
+  }
+
+  Wrapper = ({ children }) => (
```

Comment:
What do you think if to rename ```Wrapper``` -> ```RefWrapper```? I saw first a usage of it ```<this.Wrapper>``` and was a bit confused what is it about

Reply:
renamed to RefProvidingWrapper :)